src/core/auth/qgsauthorizationsettings.h

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************
+    qgsauthorizationsettings.h
+    ---------------------
+    begin                : February 2016
+    copyright            : (C) 2016 by Even Rouault
+    email                : even.rouault at spatialys.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSAUTHORIZATIONSETTINGS_H
+#define QGSAUTHORIZATIONSETTINGS_H
+
+#include "qgsauthmanager.h"
+#include "qgsapplication.h"
+#include "qgshttpheaders.h"
+
+#include <QString>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+
+#define SIP_NO_FILE
+
+struct QgsAuthorizationSettings
```

Comment:
Can you add some docs for the class itself?

Reply:
@nyalldawson thanks I added some. Do I need to use the macro `CORE_EXPORT` ? The struct is used in some of the providers